src/ComponentRegister.ts

```diff
@@ -0,0 +1,45 @@
+import { ElementType, ReactNode } from "react";
+
+export interface LinkComponentProps {
+  href: string;
+  className?: string;
+  children: ReactNode;
+}
+
+export type LinkComponent = ElementType<LinkComponentProps>;
+
+export type ComponentRegisterInitOptions = {
```

Comment:
```suggestion
export interface ComponentRegisterOptions {
  components?: {
    Link?: LinkComponent;
  }
};
```

I think it does actually make sense to scope this to components. If we end up with other non-component optons, we'll probably want a new object for those.